### PR TITLE
feat: Tidy up reading query string for mocking

### PR DESCRIPTION
**Pull request by nickrandolph**
## PR Type

What kind of change does this PR introduce?
- Code style update (formatting)
- Refactoring (no functional changes, no api changes)

## What is the current behavior?

Custom logic to read query string for wasm - looks for "mock=true"

## What is the new behavior?

Query string arguments are added to host configuration so can be accessed by configuration anywhere in the app
New mock query string is "ToDoApp:Mock=true"

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by nickrandolph** [comment]
@NVLudwig just a heads up that if/when this PR goes through the syntax for enabling mocks will be different - use "?ToDoApp:Mock=true" instead of "?mock=true" to enable mocking

**Review by carldebilly — DISMISSED** [review]

File: src/ToDo.UI/App.xaml.host.cs [hunk 1]
```diff
@@ -1,34 +1,30 @@
...
+				// TODO: Remove when the uno.extensions update comes through that includes this
+				.ConfigureHostConfiguration(config =>
+				{
+					var href = WebAssemblyRuntime.InvokeJS("window.location.href");
+					var query = new Uri(href).Query;
+						var queriesValues = System.Web.HttpUtility.ParseQueryString(query);
+						var queryDict = (from k in queriesValues.AllKeys
+										 select new { Key = k, Value = queriesValues[k] }).ToDictionary(x => x.Key, x => x.Value);
+						config.AddInMemoryCollection(queryDict);
+				})
```
Comment (carldebilly) [inline src/ToDo.UI/App.xaml.host.cs:24]: Aaammmazing

**Review by carldebilly — APPROVED** [review]

File: src/When_SmokeTest_log_2022-05-12-12-24-52.982.txt [hunk 2]
```diff
@@ -0,0 +1,85 @@
+12/05/2022 2:24:47 AM/Info: http://localhost:5000/package_98fd4b76795f720b455ac3570137d09d0d0b1389/uno-bootstrap.js 10:57 "Setting UNO_BOOTSTRAP_MONO_RUNTIME_MODE=Interpreter"
```
Comment (agneszitte-nventive) [inline src/When_SmokeTest_log_2022-05-12-12-24-52.982.txt:1]: You can remove this file I think

**Review by agneszitte-nventive — COMMENTED** [review]
@nickrandolph please update the links in the ReadMe.md file for the mock query string

**Comment by agneszitte-nventive** [comment]
Nice work!

File: src/ToDo.UI/App.xaml.host.cs (hunk 1 above)
Comment (nickrandolph) [reply src/ToDo.UI/App.xaml.host.cs]: Done

File: src/When_SmokeTest_log_2022-05-12-12-24-52.982.txt (hunk 2 above)
Comment (nickrandolph) [reply src/When_SmokeTest_log_2022-05-12-12-24-52.982.txt]: Done

**Review by agneszitte-nventive — DISMISSED** [review]

**Review by agneszitte-nventive — APPROVED** [review]

**Pull request merged**